lang/extract_json_strings.py

```diff
@@ -807,6 +807,12 @@ def extract_snippet(state, item):
         else:
             writestr(state, snip["text"])
 
+def extract_weapon_category(item):
+    outfile = get_outfile("weapon_category")
+    name = item.get("name")
+    comment = "weapon category name"
+    writestr(outfile, name, comment=comment)
```

Comment:
The script has been modified in BN, and the extraction functions use a different output algorithm.

```suggestion

def extract_weapon_category(state, item):
    name = item["name"]
    comment = "weapon category name"
    writestr(state, name, comment=comment)
```